### PR TITLE
Haiku build fix proposal.

### PR DESCRIPTION
**Pull request by devnexen**
No api to avoid address chunks to be paged out in the userland.

### CODE REVIEW

File: src/lib/ringbuffer.cpp [hunk 1]
```diff
@@ -25,6 +25,11 @@
 
 #ifdef USE_MLOCK
 	#include <sys/mman.h>
+	#ifdef __HAIKU__
```
Comment (JohannesLorenz) [inline src/lib/ringbuffer.cpp]: Why do you include `<sys/mman.h>` for Haiku at all?

Also, can't you disable `USE_MLOCK` in `cmake/process_project.txt` instead? The way it is now, `cmake/process_project.txt` will print that mlock is being used on Haiku.

**Review by JohannesLorenz — CHANGES_REQUESTED** [review]
I added one comment to your code.

File: src/lib/ringbuffer.cpp (hunk 1 above)
Comment (devnexen) [reply src/lib/ringbuffer.cpp]: good point. hope it suits better now.

**Comment by JohannesLorenz** [comment]
Just for me to understand the fix: Haiku has `sys/mman.h`, but no `mlock` in it? Normally it should be in there, this is really confusing.

**Comment by devnexen** [comment]
the man page says it is present but in fact it s not implemented :-)

**Comment by JohannesLorenz** [comment]
Many thanks for the PR! I tested it, it seems to work.

**Pull request merged**